### PR TITLE
Update the readme to use trial

### PR DESCRIPTION
**Pull request by richvdh**
- something of a corrolary to #1684

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -770,16 +779,16 @@ to install using pip and a virtualenv::
...
+    pip install lxml mock
 
 This will run a process of downloading and installing all the needed
 dependencies into a virtual env.
 
 Once this is done, you may wish to run Synapse's unit tests, to
 check that everything is installed as it should be::
 
-    python setup.py test
+    PYTHONPATH="" trial tests
```
Comment (erikjohnston) [inline README.rst]: Should this not be `PYTHONPATH="." trial tests`?

File: README.rst [hunk 2]
```diff
@@ -770,16 +779,16 @@ to install using pip and a virtualenv::
 
     virtualenv env
     source env/bin/activate
-    python synapse/python_dependencies.py | xargs -n1 pip install
-    pip install setuptools_trial mock
+    python synapse/python_dependencies.py | xargs pip install
+    pip install lxml mock
```
Comment (erikjohnston) [inline README.rst]: `lxml` requires one of the xml parser libraries, i forget its name.

Comment (richvdh) [inline README.rst]: a python lib, or a C one? libxslt1-dev is already mentioned in "Installing from source".

File: README.rst (hunk 1 above)
Comment (richvdh) [inline README.rst]: good spot

File: README.rst (hunk 2 above)
Comment (NegativeMjark) [inline README.rst]: The libxslt1-dev one I think? Might be worth mentioning it again here.

Comment (richvdh) [inline README.rst]: well, this is why I added the "make sure you have the system dependencies installed" paragraph. We might as well duplicate the whole section, and I'm super unkeen on that.

Comment (richvdh) [inline README.rst]: (should we also recommend installing python-dev? what if your distro calls it something else? argh)

File: README.rst (hunk 1 above)
Comment (erikjohnston) [inline README.rst]: On the other hand, it also seems to work :s

**Comment by richvdh** [comment]
@erikjohnston @NegativeMjark ptal?

**Pull request merged**